### ISSUE TITLE
me

### ISSUE THREAD

**Issue by NameLacker** [NameLacker/First_temp#1]
import mxnet as mx
import numpy as np
import os
import gzip
import struct
import logging

logging.getLogger().setLevel(logging.DEBUG)

def read_data(label_url, image_url):
	with gzip.open(label_url) as flbl:
		magic, num = struct.unpack(">II", flbl.read(8))
		label = np.fromstring(flbl.read(), dtype=np.int8)
	with gzip.open(image_url, 'rb') as fimg:
		magic, num, rows, cols = struct.unpack(">IIII", fimg.read(16))
		image = np.fromstring(fimg.read(), dtype=np.uint8)
		image = image.reshape(len(label), 1, rows, cols)
		image = image.astype(np.float32)/255.0
	return (label, image)

(train_lbl, train_img) = read_data('train-labels-idx1-ubyte.gz', 'train-images-idx3-ubyte.gz')
(val_lbl, val_img) = read_data('t10k-labels-idx1-ubyte.gz', 't10k-images-idx3-ubyte.gz')

batch_size = 30

train_iter = mx.io.NDArrayIter(train_img, train_lbl, batch_size, shuffle=True)
val_iter = mx.io.NDArrayIter(val_img, val_lbl, batch_size)

data = mx.symbol.Variable('data')

flatten = mx.sym.Flatten(data=data, name="flatten")

msf = mx.sym.FullyConnected
msa = mx.sym.Activation

fc1 = msf(data=flatten, num_hidden=128, name="fc1")
act1 = msa(data=fc1, act_type="relu", name="act1")

fc2 = msf(data=act1, num_hidden=64, name="fc2")
act2 = msa(data=fc2, act_type="relu", name="act2")

fc3 = msf(data=act2, num_hidden=10, name="fc3")
net = mx.sym.SoftmaxOutput(data=fc3, name='softmax')

module = mx.mod.Module(symbol=net, context=mx.cpu(0))

module.fit(
	train_iter,
	eval_data=val_iter,
	optimizer = 'sgd',
	optimizer_params = {'learning_rate' : 0.2, 'lr_scheduler' : mx.lr_scheduler.FactorScheduler(step=60000/batch_size, factor=0.9)},
	num_epoch = 20,
	batch_end_callback = mx.callback.Speedometer(batch_size, 60000/batch_size)
)
